module/batch-action/src/Infrastructure/Handler/ReprocessBatchActionCommandHandler.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\BatchAction\Infrastructure\Handler;
+
+use Ergonode\BatchAction\Domain\Repository\BatchActionRepositoryInterface;
+use Ergonode\BatchAction\Domain\Entity\BatchAction;
+use Ergonode\BatchAction\Domain\Command\ReprocessBatchActionCommand;
+use Webmozart\Assert\Assert;
+use Ergonode\BatchAction\Domain\ValueObject\BatchActionStatus;
+
+class ReprocessBatchActionCommandHandler
+{
+    private BatchActionRepositoryInterface $repository;
+
+    public function __construct(BatchActionRepositoryInterface $repository)
+    {
+        $this->repository = $repository;
+    }
+
+    public function __invoke(ReprocessBatchActionCommand $command): void
+    {
+        $batchAction = $this->repository->load($command->getId());
+        Assert::isInstanceOf($batchAction, BatchAction::class);
+        $batchAction->setPayload($command->getPayload());
+        $batchAction->setStatus(new BatchActionStatus(BatchActionStatus::PROCESSED));
```

Comment:
```suggestion
        $batchAction->setStatus(new BatchActionStatus(BatchActionStatus::PRECESSED));
```
it seems like a typo in status name - I think we do not have PROCESSED one